### PR TITLE
Use dependencies for yaml files from uap-core

### PR DESCRIPTION
**Pull request by douglaspalmer**
We would like to use uap-java in the keycloak project (https://www.keycloak.org). However, to do so I need to make a change to your build system. Would my proposed changes be acceptable to you?

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -93,6 +66,38 @@
...
+                  <classifier>tests</classifier>
+                  <type>test-jar</type>
+                  <outputDirectory>${project.build.directory}/test-classes</outputDirectory>
+                  <includes>**/*.yaml</includes>
+                </artifactItem>
+              </artifactItems>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin-->
```
Comment (bpossolo) [inline pom.xml]: why is this all commented out?

**Comment by bpossolo** [comment]
i think this change makes sense.
i always thought the `<resource>` blocks with references to `{baseDir}` kind of odd and unconventional for maven

File: pom.xml (hunk 1 above)
Comment (douglaspalmer) [reply pom.xml]: Sorry, commented it out to see if unpacking was necessary. It wasn't so I will delete it and update the pull request.

**Comment by abstractj** [comment]
@bpossolo do you think we can get it in further releases? Is there anything missing?

**Comment by ThoSap** [comment]
@bpossolo when will this be merged into master?
This would be awesome and very much appreciated by the community! 🚀🤩

**Comment by bpossolo** [comment]
Ok. I’ll merge and just note that we should remove the git module reference to the other project

**Comment by bpossolo** [comment]
I just gave this PR a spin and it doesn't work.

There is no artifact in Maven central that matches:
* group id: com.github.ua-parser
* artifact id: uap-core

I presume this was never tested or you created the artifact yourself and installed it in your local Maven repo so I'm going to revert.

**Comment by douglaspalmer** [comment]
You need to apply the PR for core too.

**Comment by douglaspalmer** [comment]
The PR is well tested and works; you just need to apply the other half: https://github.com/ua-parser/uap-core/pull/348

**Comment by bpossolo** [comment]
Oh I see. It would have been good to link this PR to [that one](https://github.com/ua-parser/uap-core/pull/348) and made clear there was a dependency on that.

I don't have control of uap-core... so it will require that to be done first.

**Comment by bpossolo** [comment]
Out of curiosity, can you explain what is preventing you from using the uap-java artifact directly which includes the regexp yaml files?

**Comment by douglaspalmer** [comment]
> Out of curiosity, can you explain what is preventing you from using the uap-java artifact directly which includes the regexp yaml files?

For my use case; we build everything from source internally. We don't use external binaries.

**Comment by bpossolo** [comment]
Ok, I thought that might be that case.
Are you aware that the Parser class has a constructor which accepts an InputStream to the regexp file?
In the meantime, can you get the uap-core repo + it's yaml file then use that constructor instead?

I presume the maven tests will all fail without uap-core where it's expected to be so you might need to add `-Dmaven.test.skip=true` when building the artfiact.

**Comment by bpossolo** [comment]
I commented in the other PR asking @commenthol  for advice.
I can't just merge your PR into his project as I don't really consider myself a maintainer for that.

If I remember correctly, I'll need to go through all the steps to get a new artifact published to Maven Central which requires some time as it's a bit of a manual back-and-forth process.

**Pull request merged**